src/btree/bt_curnext.c

```diff
@@ -101,7 +105,13 @@ __cursor_fix_next(WT_CURSOR_BTREE *cbt, bool newpage)
 	    cbt->ins_head, cbt->ins_stack, cbt->next_stack, cbt->recno);
 	if (cbt->ins != NULL && cbt->recno != WT_INSERT_RECNO(cbt->ins))
 		cbt->ins = NULL;
-	upd = cbt->ins == NULL ? NULL : __wt_txn_read(session, cbt->ins->upd);
+
+	upd = NULL;			/* -Werror=maybe-uninitialized */
```

Comment:
This initialization should go in the in general initialization block below the declaration block (unless it's in a loop).